string location
string message
---
int32 result
